searx/engines/wordnik.py

```diff
@@ -0,0 +1,129 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+ Wordnik (general)
+"""
+
+import re
+from lxml.html import fromstring
+from searx.utils import extract_text
+from searx.raise_for_httperror import raise_for_httperror
+
+# about
+about = {
+    "website": 'https://www.wordnik.com',
+    "wikidata_id": 'Q8034401',
+    "official_api_documentation": None,
+    "use_official_api": False,
+    "require_api_key": False,
+    "results": 'HTML',
+}
+
+categories = ['general']
+paging = False
+
+URL = 'https://www.wordnik.com'
+SEARCH_URL = URL + '/words/{query}'
+
+
+def request(query, params):
+    params['url'] = SEARCH_URL.format(query=query)
+    return params
+
+
+def response(resp):
+    if resp.status_code == 404:
+        return []
+
+    try:
+        dom = fromstring(resp.text)
+    except Exception:
+        raise_for_httperror(resp)
+        return[]
+
+    results = []
+    word_defs = []
+    word_defs_ib = []
+    urls = []
+    lines = ""
+    lines_ib = "<div>"
+
+    word_xpath = '//*[@id="headword"]/text()'
+    from_xpath = '//*[@id="define"]/div/h3[1]'
+    definitions_xpath = '//*[@id="define"]/div/ul[1]'
+    misspelling_xpath = '//*[@id="define"]/div/ul/li/xref'
+
+    word = extract_text(dom.xpath(word_xpath))
+    definition_from = extract_text(dom.xpath(from_xpath))
+    definitions = dom.xpath(definitions_xpath)
+
+    if len(definitions) == 0:
+        return []
+
+    for definition in definitions:
+        for i in range(len(definition)):
+            definition_extra_ib = ""
+            definition_extra = ""
+
+            part_of_speach_xpath = f"./li[{i+1}]/abbr"
+            definition_extra_xpath = f"./li[{i+1}]/i"
+            definition_text_xpath = f"./li[{i+1}]"
+
+            part_of_speach = extract_text(definition.xpath(part_of_speach_xpath))
```

Comment:
Nit: isn't `speach` supposed to be `speech`?

Reply:
Now you know why do I need wordnik :-)